## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [2]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [4]:
# Write your code for importing the csv file here
movies = pd.read_csv('Movies.csv')
movies

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ... num_user_for_reviews language  \
0         Action|Adventure|Fantasy|Sci-Fi  ...               3054.0  English   
1                Action|Adventure|Fantasy  ...               1238.0  English   
2               Action|Adventure|Thriller  ...                994.0  English   
3                         Action|Thriller  ...               2701.0  English   
4                 Action|Adventure|Sci-Fi  ...                738.0  English   
...                                   ...  ...                  ...      ...   
3848                Drama|Sci-Fi|Thriller  ...                371.0  English   
3849                             Thriller  ...                 35.0  English   
3850  Action|Crime|Drama|Romance|Thriller  ...                130.0  Spanish   
3851                         Comedy|Drama  ...                 14.0  English   
3852                          Documentary  ...                 84.0  English   

          country  content_rating       budget  title_year  \
0             USA           PG-13  237000000.0      2009.0   
1             USA           PG-13  300000000.0      2007.0   
2              UK           PG-13  245000000.0      2015.0   
3             USA           PG-13  250000000.0      2012.0   
4             USA           PG-13  263700000.0      2012.0   
...           ...             ...          ...         ...   
3848          USA           PG-13       7000.0      2004.0   
3849  Philippines       Not Rated       7000.0      2005.0   
3850          USA               R       7000.0      1992.0   
3851          USA       Not Rated       9000.0      2011.0   
3852          USA              PG       1100.0      2004.0   

     actor_2_face

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [14]:
# Write your code for inspection here
movies.info()<3853

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   color                      3851 non-null   object 
 1   director_name              3853 non-null   object 
 2   num_critic_for_reviews     3852 non-null   float64
 3   duration                   3852 non-null   float64
 4   director_facebook_likes    3853 non-null   float64
 5   actor_3_facebook_likes     3847 non-null   float64
 6   actor_2_name               3852 non-null   object 
 7   actor_1_facebook_likes     3853 non-null   float64
 8   gross                      3853 non-null   float64
 9   genres                     3853 non-null   object 
 10  actor_1_name               3853 non-null   object 
 11  movie_title                3853 non-null   object 
 12  num_voted_users            3853 non-null   int64  
 13  cast_total_facebook_likes  3853 non-null   int64

TypeError: '<' not supported between instances of 'NoneType' and 'int'

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [22]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations

movies.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   director_name           3853 non-null   object 
 1   num_critic_for_reviews  3852 non-null   float64
 2   gross                   3853 non-null   float64
 3   genres                  3853 non-null   object 
 4   actor_1_name            3853 non-null   object 
 5   movie_title             3853 non-null   object 
 6   num_voted_users         3853 non-null   int64  
 7   num_user_for_reviews    3853 non-null   float64
 8   language                3850 non-null   object 
 9   budget                  3853 non-null   float64
 10  title_year              3853 non-null   float64
 11  imdb_score              3853 non-null   float64
 12  movie_facebook_likes    3853 non-null   int64  
dtypes: float64(6), int64(2), object(5)
memory usage: 391.4+ KB


#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'. 

In [62]:
# Write you code here
round(100*(movies.isnull().sum()/len(movies.index)), 2)
movies.head()

director_name  num_critic_for_reviews       gross  \
0      James Cameron                   723.0  760.505847   
1     Gore Verbinski                   302.0  309.404152   
2         Sam Mendes                   602.0  200.074175   
3  Christopher Nolan                   813.0  448.130642   
4     Andrew Stanton                   462.0   73.058679   

                            genres     actor_1_name  \
0  Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1         Action|Adventure|Fantasy      Johnny Depp   
2        Action|Adventure|Thriller  Christoph Waltz   
3                  Action|Thriller        Tom Hardy   
4          Action|Adventure|Sci-Fi     Daryl Sabara   

                                 movie_title  num_voted_users  \
0                                    Avatar            886204   
1  Pirates of the Caribbean: At World's End            471220   
2                                   Spectre            275868   
3                     The Dark Knight Rises           1144337   
4                               John Carter            212204   

   num_user_for_reviews language    budget  title_year  imdb_score  \
0                3054.0  English  0.000237      2009.0         7.9   
1                1238.0  English  0.000300      2007.0         7.1   
2                 994.0  English  0.000245      2015.0         6.8   
3                2701.0  English  0.000250      2012.0         8.5   
4                 738.0  English  0.000264      2012.0         6.6   

   movie_facebook_likes  
0                 33000  
1                     0  
2                 85000  
3                164000  
4                 24000

#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [55]:
# Write your code for filling the NaN values in the 'language' column here
movies["language"].replace({np.nan:'English'}, inplace=True)
movies[movies[["language"]]=="English"]
movies['language'].value_counts()

English       3674
French          37
Spanish         26
Mandarin        14
German          13
Japanese        12
Hindi           10
Cantonese        8
Italian          7
Korean           5
Portuguese       5
Norwegian        4
Persian          3
Danish           3
Dutch            3
Thai             3
Dari             2
Hebrew           2
Indonesian       2
Aboriginal       2
Aramaic          1
Icelandic        1
Hungarian        1
Mongolian        1
Zulu             1
Swedish          1
Bosnian          1
Maya             1
Arabic           1
Romanian         1
Filipino         1
Dzongkha         1
Vietnamese       1
Telugu           1
Russian          1
Kazakh           1
None             1
Czech            1
Name: language, dtype: int64

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [61]:
# Write your code for unit conversion here
movies["budget"]=movies["budget"]/1000000
movies["gross"]=movies["gross"]/1000000

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [64]:
# Write your code for creating the profit column here
movies["Profit"] = movies['gross'] - movies['budget']

In [68]:
# Write your code for sorting the dataframe here
movies.sort_values('Profit',ascending=False,inplace=True)

In [72]:
# Write your code to get the top 10 profiting movies here
movies.head(10)

director_name  num_critic_for_reviews       gross  \
0         James Cameron                   723.0  760.505847   
25        James Cameron                   315.0  658.672302   
28      Colin Trevorrow                   644.0  652.177271   
16          Joss Whedon                   703.0  623.279547   
64    Christopher Nolan                   645.0  533.316061   
230        George Lucas                   320.0  474.544677   
2704       George Lucas                   282.0  460.935665   
7           Joss Whedon                   635.0  458.991599   
3     Christopher Nolan                   813.0  448.130642   
552      Andrew Adamson                   205.0  436.471036   

                                                 genres         actor_1_name  \
0                       Action|Adventure|Fantasy|Sci-Fi          CCH Pounder   
25                                        Drama|Romance    Leonardo DiCaprio   
28                     Action|Adventure|Sci-Fi|Thriller  Bryce Dallas Howard   
16                              Action|Adventure|Sci-Fi      Chris Hemsworth   
64                          Action|Crime|Drama|Thriller       Christian Bale   
230                     Action|Adventure|Fantasy|Sci-Fi      Natalie Portman   
2704                    Action|Adventure|Fantasy|Sci-Fi        Harrison Ford   
7                               Action|Adventure|Sci-Fi      Chris Hemsworth   
3                                       Action|Thriller            Tom Hardy   
552   Adventure|Animation|Comedy|Family|Fantasy|Romance       Rupert Everett   

                                     movie_title  num_voted_users  \
0                                        Avatar            886204   
25                                      Titanic            793059   
28                               Jurassic World            418214   
16                                 The Avengers            995415   
64                              The Dark Knight           1676169   
230   Star Wars: Episode I - The Phantom Menace            534658   
2704         Star Wars: Episode IV - A New Hope            911097   
7                       Avengers: Age of Ultron            462669   
3                         The Dark Knight Rises           1144337   
552                                     Shrek 2            314630   

      num_user_for_reviews language    budget  title_year  imdb_score  \
0                   3054.0  English  0.000237      2009.0         7.9   
25                  2528.0  English  0.000200      1997.0         7.7   
28                  1290.0  English  0.000150      2015.0         7.0   
16                  1722.0  English  0.000220      2012.0         8.1   
64                  4667.0  English  0.000185      2008.0         9.0   
230                 3597.0  English  0.000115      1999.0         6.5   
2704                1470.0  English  0.000011      1977.0         8.7   
7                   1117.0  English  0.000250      2015.0         7.5   
3                   2701.0  English  0.000250      2012.0         8.5   
552                  483.0  English  0.000150      2004.0         7.2   

      movie_facebook_likes      Profit  
0                    33000  760.505610  
25                   26000  658.672102  
28                  150000  652.177121  
16                  123000  623.279327  
64                   37000  533.315876  
230                  13000  474.544562  
2704                 33000  460.935654  
7                   118000  458.991349  
3                   164000  448.130392  
552                      0  436.470886

**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000. 

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [96]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe 
# and name that dataframe as 'IMDb_Top_250'
IMDb_Top_250 = movies.sort_values('Profit',ascending=False).head(250)
((IMDb_Top_250.imdb_score>8.5) & (IMDb_Top_250.imdb_score<9)).sum()

10

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [101]:
# Write your code for creating three new dataframes here
Meryl_Streep = movies.loc[movies.actor_1_name=="Meryl Streep"]
Meryl_Streep

director_name  num_critic_for_reviews       gross  \
1322   David Frankel                   208.0  124.732962   
392     Nancy Meyers                   187.0  112.703470   
1132     Nora Ephron                   252.0   94.125426   
1471  Sydney Pollack                    66.0   87.100000   
1514   David Frankel                   234.0   63.536011   
1038   Curtis Hanson                    42.0   46.815748   
1784  Stephen Daldry                   174.0   41.597830   
2500  Phyllida Lloyd                   331.0   29.959436   
1563   Carl Franklin                    64.0   23.209440   
2793   Robert Altman                   211.0   20.338609   
1390  Robert Redford                   227.0   14.998070   

                               genres  actor_1_name  \
1322             Comedy|Drama|Romance  Meryl Streep   
392              Comedy|Drama|Romance  Meryl Streep   
1132          Biography|Drama|Romance  Meryl Streep   
1471          Biography|Drama|Romance  Meryl Streep   
1514             Comedy|Drama|Romance  Meryl Streep   
1038  Action|Adventure|Crime|Thriller  Meryl Streep   
1784                    Drama|Romance  Meryl Streep   
2500          Biography|Drama|History  Meryl Streep   
1563                            Drama  Meryl Streep   
2793               Comedy|Drama|Music  Meryl Streep   
1390               Drama|Thriller|War  Meryl Streep   

                    movie_title  num_voted_users  num_user_for_reviews  \
1322     The Devil Wears Prada            286178                 631.0   
392           It's Complicated             69860                 214.0   
1132             Julie & Julia             79264                 277.0   
1471             Out of Africa             52339                 200.0   
1514              Hope Springs             34258                 178.0   
1038            The River Wild             32544                  69.0   
1784                 The Hours            102123                 660.0   
2500             The Iron Lady             82327                 350.0   
1563            One True Thing              9283                 112.0   
2793  A Prairie Home Companion             19655                 280.0   
1390           Lions for Lambs             41170                 298.0   

     language    budget  title_year  imdb_score  movie_facebook_likes  \
1322  English  0.000035      2006.0         6.8                     0   
392   English  0.000085      2009.0         6.6                     0   
1132  English  0.000040      2009.0         7.0                 13000   
1471  English  0.000031      1985.0         7.2                     0   
1514  English  0.000030      2012.0         6.3                     0   
1038  English  0.000045      1994.0         6.3                     0   
1784  English  0.000025      2002.0         7.6                     0   
2500  English  0.000013      2011.0         6.4                 18000   
1563  English  0.000030      1998.0         7.0                   592   
2793  English  0.000010      2006.0         6.8                   683   
1390  English  0.000035      2007.0         6.2                     0   

          Profit  
1322  124.732927  
392   112.703385  
1132   94.125386  
1471   87.099969  
1514   63.535981  
1038   46.815703  
1784   41.597805  
2500   29.959423  
1563   23.209410  
2793   20.338599  
1390   14.998035

In [102]:
Leo_Caprio = movies.loc[movies.actor_1_name=="Leonardo DiCaprio"]

In [103]:
Brad_Pitt = movies.loc[movies.actor_1_name=="Brad Pitt"]

In [104]:
# Write your code for combining the three dataframes here
Combined = pd.concat([Meryl_Streep,Leo_Caprio,Brad_Pitt])

In [105]:
# Write your code for grouping the combined dataframe here
Combined

director_name  num_critic_for_reviews       gross  \
1322          David Frankel                   208.0  124.732962   
392            Nancy Meyers                   187.0  112.703470   
1132            Nora Ephron                   252.0   94.125426   
1471         Sydney Pollack                    66.0   87.100000   
1514          David Frankel                   234.0   63.536011   
1038          Curtis Hanson                    42.0   46.815748   
1784         Stephen Daldry                   174.0   41.597830   
2500         Phyllida Lloyd                   331.0   29.959436   
1563          Carl Franklin                    64.0   23.209440   
2793          Robert Altman                   211.0   20.338609   
1390         Robert Redford                   227.0   14.998070   
25            James Cameron                   315.0  658.672302   
94        Christopher Nolan                   642.0  292.568851   
173   Alejandro G. Iñárritu                   556.0  183.635922   
859        Steven Spielberg                   194.0  164.435221   
283       Quentin Tarantino                   765.0  162.804648   
49             Baz Luhrmann                   490.0  144.812796   
3025           Baz Luhrmann                   490.0  144.812796   
347         Martin Scorsese                   352.0  132.373442   
431         Martin Scorsese                   490.0  127.968405   
294         Martin Scorsese                   606.0  116.866727   
246         Martin Scorsese                   267.0  102.608827   
312         Martin Scorsese                   233.0   77.679638   
293            Edward Zwick                   166.0   57.366262   
1336        Randall Wallace                    83.0   56.876365   
2484           Baz Luhrmann                   106.0   46.338728   
934             Danny Boyle                   118.0   39.778599   
608            Ridley Scott                   238.0   39.380442   
1364         Clint Eastwood                   392.0   37.304950   
1046             Sam Mendes                   323.0   22.877808   
1457              Sam Raimi                    63.0   18.636537   
1909             Jerry Zaks                    45.0   12.782508   
244              Doug Liman                   233.0  186.336103   
383       Steven Soderbergh                   186.0  183.405771   
142       Wolfgang Petersen                   220.0  133.228348   
97            David Fincher                   362.0  127.490802   
243       Steven Soderbergh                   198.0  125.531634   
887             Neil Jordan                   120.0  105.264608   
448              David Ayer                   406.0   85.707116   
579     Jean-Jacques Annaud                    76.0   37.901509   
646           David Fincher                   315.0   37.023395   
2030  Alejandro G. Iñárritu                   285.0   34.300771   
749         Patrick Gilmore                    98.0   26.288320   
2428         Andrew Dominik                   414.0   14.938570   
1397        Terrence Malick                   584.0   13.303319   
2602             Tony Scott                   122.0   12.281500   
1608         Andrew Dominik                   273.0    3.904982   
2146    Angelina Jolie Pitt                   131.0    0.531009   
367              Tony Scott                   142.0    0.026871   

                                                 genres       actor_1_name  \
1322                               Comedy|Drama|Romance       Meryl Streep   
392                                Comedy|Drama|Romance       Meryl Streep   
1132                            Biography|Drama|Romance       Meryl Streep   
1471                            Biography|Drama|Romance       Meryl Streep   
1514                               Comedy|Drama|Romance       Meryl Streep   
1038                    Action|Adventure|Crime|Thriller       Meryl Streep   
1784                                      Drama|Romance       Meryl Streep   
2500                            Biograp

In [115]:
# Write the code for finding the mean of critic reviews and audience reviews here
Combined['num_critic_for_reviews'].mean()
Combined['num_user_for_reviews'].mean()
Combined_by_segment = Combined.groupby('actor_1_name')

Combined_by_segment['num_user_for_reviews'].mean()
Combined_by_critics = Combined.groupby('actor_1_name')

Combined_by_critics['num_critic_for_reviews'].mean()

actor_1_name
Brad Pitt            245.000000
Leonardo DiCaprio    330.190476
Meryl Streep         181.454545
Name: num_critic_for_reviews, dtype: float64

#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [111]:
Meryl_Streep['num_user_for_reviews'].mean()


297.1818181818182

In [112]:
Leo_Caprio['num_user_for_reviews'].mean()


914.4761904761905

In [113]:
Brad_Pitt['num_user_for_reviews'].mean()

742.3529411764706